app/components/search.js

```diff
@@ -1,8 +1,52 @@
-import React, {Component} from 'react'
+import React, {Component, PropTypes} from 'react'
+import {searchTracks} from '../resource'
+import SearchResults from './search-results'
+import _ from 'lodash'
 
 class Search extends Component {
+  static propTypes = {
+    actions: PropTypes.object,
+    tracksList: PropTypes.array
+  }
+
+  state = {
+    searchQuery: '',
+    error: '',
+    showResults: false,
+    titles: []
+  }
+
+  handleSearch = _.debounce((query) => {
+    const {refreshSearchResults} = this.props.actions
+    if (query.trim().length === 0) {
+      return refreshSearchResults([])
+    }
+    return searchTracks(query)
+    .then(response => {
+      refreshSearchResults(response)
+      response.length && this.setState({showResults: true})
+    })
+    .catch(error => {
+      this.setState({error: error.message})
+    })
+  }, 800)
+
+  inputHandler = event => {
```

Comment:
rename into handleInput (keep one style of naming, either inputHandler, searchHandler or handleInput, handleSearch)